Sources/Differ/Diff+AppKit.swift

```diff
@@ -51,11 +105,12 @@ extension NSTableView {
         )
     }
 
+    @available(*, deprecated, message: "Use `apply(_ patches: …)` based on `ExtendedPatch` instead. Deprecated because it has errors animating multiple moves.")
     public func apply(
         _ diff: ExtendedDiff,
         deletionAnimation: NSTableView.AnimationOptions = [],
         insertionAnimation: NSTableView.AnimationOptions = [],
-        indexPathTransform: (IndexPath) -> IndexPath = { $0 }
+        indexPathTransform: (IndexPath) -> IndexPath
```

Comment:
Removing the default parameter value here is going to break clients relying on it. For the other methods, I guess calls typecheck to the new, fixed versions in case the argument is omitted.

Why are me moving away from `IndexPath`s to one-dimensional indices though? Is this a macOS/`NSTableView` thing?

Reply:
My idea is:
- ideally previous callers should be migrated to the new implementation which doesn't have the solved bug, and they don't even need to know about it. I'm confident that the initial bugs with regard to moves is fixed, and nobody should need to update the call to take advantage of that. unless: 
- I left the indexPathTransform in there because for cases where there is a non-identity transform defined, it is not immediately clear how to map this to the actually only row-based approach. People who declare an indexPathTransform will then meet the deprecation warning and can transition

macOS NSTableView doesn't have sections like UITableView, and operates on `row: Int` when addressing rows. Since we only operating on rows (not on columns, which NSTableView supports),  `IndexPath`'s typical two dimensions don't translate to rows in a  meaningful way

Reply:
Okay, let me rephrase. As far as I understood, the bug you fixed originated from using `ExtendedDiff` vs `ExtendedPatch<T>`.

We therefore cannot fix `apply(_: ExtendedDiff, ...)` and should deprecate this method as you did. However, I'd leave the default parameter value so that users relying on the default parameter actually see the deprecation instead of getting a compiler error.

In contrast, we _can_ fix the two `animateRowChanges(...)` methods without changing their signature, meaning all users get to benefit from the fix without having to change anything. If you are confident that the new behavior is correct, that's what we should aim for. However, with your current changes, only users who don't use `indexPathTransform` would get the fix.

Can we keep the `indexPathTransform` parameters, or did I miss something and they are related to the bug fix as well? Of course you could then add the `rowIndexTransform`-versions as an overlay to those, though I'd refrain from deprecating the `indexPathTransform`-based ones.

Reply:
Ah yes got it now. You're right, on the `apply(…diff:…)` method removing the default param doesn't help.
With regard to the indexPathTransform: They are not related to the bugfix, it just confused me for a bit when I was confronted with them since they don't mean too much in the NSTableView.

Reply:
I concur that the default parameter should be restored here, unless it's not required for some reason — I don't mind breaking changes when they are warranted, but this seems like unnecessary breakage to me, doubly so given that this method is deprecated.